server/src/main/java/org/elasticsearch/action/support/DestructiveOperations.java

```diff
@@ -59,6 +61,9 @@ public void failDestructive(String[] aliasesOrIndices) {
             if (hasWildcardUsage(aliasesOrIndices[0])) {
                 throw new IllegalArgumentException("Wildcard expressions or all indices are not allowed");
             }
+        } else if (Arrays.equals(aliasesOrIndices, new String[]{"*", "-*"})) {
```

Comment:
Can you make the String[] a static final variable?

Reply:
Also you could simplify the code and remove the else below by making the check Arrays.equals() == false. Either way is fine